File: 03-improve_exceptions_flow/3-add_exhaustivity/src/contexts/rrss/users/domain/UserDoesNotExistError.ts
import { DomainError } from "../../../shared/domain/DomainError";

export class UserDoesNotExistError extends DomainError {
	readonly type = "UserDoesNotExistError";
	readonly message = `The user ${this.id} does not exist`;

	constructor(public readonly id: string) {
		super();
	}
}

File: 07-langs_particularities/2-java/src/main/java/tv/codely/contexts/rrss/users/domain/UserDoesNotExistError.java
package tv.codely.contexts.rrss.users.domain;

import tv.codely.contexts.shared.domain.DomainError;

public class UserDoesNotExistError extends DomainError {
    public UserDoesNotExistError(String id) {
        super(String.format("The user <%s> does not exist", id));
    }
}
